modules/dop/services/pipeline/pipeline.go

```diff
@@ -397,18 +407,19 @@ func generatorWorkspaceNS(appID uint64, workspace string) ([]string, error) {
 func (p *Pipeline) generatorPipelineNS(appID uint64, branch string, workspace string) ([]string, error) {
 	var cmNamespaces []string
 	// 创建 default namespace
-	cmNamespaces = append(cmNamespaces, fmt.Sprintf("%s-%d-default", apistructs.PipelineAppConfigNameSpacePreFix, appID))
+	cmNamespaces = append(cmNamespaces, fmt.Sprintf("%s-%d-default", cms.PipelineAppConfigNameSpacePrefix, appID))
 
 	// TODO 直接使用workspace，不用映射 support hotfix
 	// hotfix support 兼容判断,如果有历史遗留参数,使用历史分支级配置 不用workspace
 	if gitflowutil.IsHotfix(branch) || gitflowutil.IsSupport(branch) {
 		branchPrefix, _ := gitflowutil.GetReferencePrefix(branch)
-		ns := fmt.Sprintf("%s-%d-%s", apistructs.PipelineAppConfigNameSpacePreFix, appID, branchPrefix)
-		configs, err := p.bdl.GetPipelineCmsNsConfigs(ns, apistructs.PipelineCmsGetConfigsRequest{
+		ns := fmt.Sprintf("%s-%d-%s", cms.PipelineAppConfigNameSpacePrefix, appID, branchPrefix)
+		configs, err := p.cms.GetCmsNsConfigs(context.Background(), &cmspb.CmsNsConfigsGetRequest{
+			Ns:             ns,
 			PipelineSource: "dice",
 		})
 		if err == nil {
-			if len(configs) > 0 {
+			if len(configs.Data) > 0 {
```

Comment:
dont know struct of configs, will it panic?

Reply:
Won't.